webextension/forceOpenPageActionPopup.js

```diff
@@ -12,17 +12,17 @@ ChromeUtils.defineModuleGetter(this, "PageActions",
 ChromeUtils.defineModuleGetter(this, "BrowserWindowTracker",
                                      "resource:///modules/BrowserWindowTracker.jsm");
 
-/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "forceOpenPageActionPopup" }] */
 this.forceOpenPageActionPopup = class extends ExtensionAPI {
   getAPI(context) {
-    const pageActionID = ExtensionUtils.makeWidgetId(context.extension.id);
+    const extension = context.extension;
+    const pageActionAPI = extension.apiManager.getAPI("pageAction", extension,
+                                                      context.envType);
     return {
       experiments: {
         pageAction: {
           async forceOpenPopup() {
-            const pageAction = PageActions.actionForID(pageActionID);
             const window = BrowserWindowTracker.getTopWindow();
-            window.BrowserPageActions.togglePanelForAction(pageAction);
+            pageActionAPI.handleClick(window);
```

Comment:
Does this open the right page action popup?

The other option would just be to dispatch a `click` event on our pageAction button.

Reply:
I first tried dispatching a click event manually like that, but it doesn't work. I think it only works in the testing code because it uses the SpecialPowers API to dispatch the event.

This is calling the same hook in the webextensions code that seems to be called to open the popup in any case, so I think it's safe (I would just be copy-pasting the code in this method otherwise to open the popup anyway, so I would rather just re-use it instead).

Reply:
Cool!